src/clients/rest/rest_client.ts

```diff
@@ -22,15 +22,17 @@ class RestClient extends HttpClient {
     }
   }
 
-  protected async request(params: RequestParams): Promise<RestRequestReturn> {
+  protected async request<T = unknown>(
+    params: RequestParams,
+  ): Promise<RestRequestReturn<T>> {
     params.extraHeaders = {
       [ShopifyHeader.AccessToken]: Context.IS_PRIVATE_APP
         ? Context.API_SECRET_KEY
         : (this.accessToken as string),
       ...params.extraHeaders,
     };
 
-    const ret = (await super.request(params)) as RestRequestReturn;
+    const ret = (await super.request<T>(params)) as RestRequestReturn<T>;
```

Comment:
Part of my previous suggestion `HttpClient#request` should also accept a type param no? and then we can just pass
```suggestion
    const ret = await super.request<RestRequestReturn>(params);
```

like this? Maybe I'm misreading something?

Reply:
So, I thought about this but I _think_ `RestRequestReturn` is at the same level as `RequestReturn`, so I think typing the `request` call would end up returning `RequestReturn<RestRequestReturn<T>>`, which isn't the goal here.

In this case I think we actually need the cast because `RestRequestReturn` is a superset of `RequestReturn` rather than a component.